atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/SimpleTransactionService.java

```diff
@@ -39,12 +39,16 @@
 
 public final class SimpleTransactionService implements EncodingTransactionService {
     private final AtomicTable<Long, Long> txnTable;
+    private final AtomicTable<Long, TransactionStatus> txnTableV2;
     private final TransactionStatusEncodingStrategy<?> encodingStrategy;
 
     private SimpleTransactionService(
-            AtomicTable<Long, Long> txnTable, TransactionStatusEncodingStrategy<?> encodingStrategy) {
+            AtomicTable<Long, Long> txnTable,
+            AtomicTable<Long, TransactionStatus> txnTableV2,
+            TransactionStatusEncodingStrategy<?> encodingStrategy) {
         this.encodingStrategy = encodingStrategy;
         this.txnTable = txnTable;
+        this.txnTableV2 = txnTableV2;
```

Comment:
For discussion: do we think we need a separate `TimestampExtractingAtomicTable` layer any more / would it make sense to perform the translations directly in this class?

Reply:
we actually have two types of timestampExtractingAtomicTables - for schemas [1, 3] and 4. Not super keen on merging until we clear out/ better place the other branch.

Reply:
Yeah, doesn't need to be here or in this PR - that's fair if we have more than one impl. I didn't really like that we have two tables that actually have the same backing store, but we don't have to clean that up now